stdlib/public/core/ShadowProtocols.swift

```diff
@@ -74,6 +74,17 @@ internal protocol _NSArrayCore: _NSCopying, _NSFastEnumeration {
   var count: Int { get }
 }
 
+// We want -objectAtIndex: to return Unmanaged<AnyObject> when called from ObjC,
+// but Swift also calls it in _CocoaArrayWrapper, and we want that to use the
+// correct type so that swiftc doesn't get confused about memory management.
+// So we have this alternate protocol we can unsafeBitCast things to that
+// declares -objectAtIndex: with a different signature
+@unsafe_no_objc_tagged_pointer @objc
+internal protocol _NSArrayCoreAlternateSignatures {
```

Comment:
@lorentey this is extremely bad, but I believe gets the right effect, and the tests appear to agree. Thoughts?

Reply:
Alternate wacky idea: change Unmanaged.takeUnretainedValue to use objc_retainAutoreleasedReturnValue instead of objc_retain.

Reply:
…possibly just when it was clearly a return value, through optimizer magic.

Reply:
Karoy came up with a less bad option :)

Reply:
Oh yeah. I guess that works.